### ISSUE TITLE
Make serde an optional dependency

### ISSUE THREAD

**Issue by uint** [ndebuhr/sim#6]
For a user using _Sim_ as a Rust library, serialization might be unnecessary and pulling in `serde`, `serde_json` and `serde_yaml` might not make sense. It would probably make sense to put all serialization/deserialization stuff behind a `serde` [feature](https://doc.rust-lang.org/cargo/reference/features.html).

**Comment by ndebuhr** [comment]
Agreed.  I'll look into this once the `Model` strategy has settled (e.g., #2) .
